### PR TITLE
Fixes the wallet alt-click so you can't take your ID out while unconscious or in soft crit

### LINKED ISSUE #51350: you can pop IDs out of your wallet while unconscious
alt+click it

### PR DESCRIPTION
**Pull request by ivanmixo**
## About The Pull Request

Title. 

## Why It's Good For The Game

Closes #51350, in other words fixes a stupid bug

## Changelog
:cl:
fix: You can no longer take your ID out while unconscious or in soft crit
/:cl:

### CODE REVIEW

File: code/datums/components/storage/concrete/wallet.dm [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /datum/component/storage/concrete/wallet/on_alt_click(datum/source, mob/user)
-	if(!isliving(user) || !user.CanReach(parent))
+	if(!isliving(user) || !user.CanReach(parent) || SOFT_CRIT || UNCONSCIOUS)
```
Comment (Tlaltecuhtli) [inline code/datums/components/storage/concrete/wallet.dm:2]: uhhh maybe you should check if something is == to those defines

Comment (Fikou) [inline code/datums/components/storage/concrete/wallet.dm:2]:
```suggestion
	if(!isliving(user) || !user.CanReach(parent) || (user.stat == SOFT_CRIT || UNCONSCIOUS))
```

File: code/datums/components/storage/concrete/wallet.dm [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /datum/component/storage/concrete/wallet/on_alt_click(datum/source, mob/user)
-	if(!isliving(user) || !user.CanReach(parent))
+	if(!isliving(user) || !user.CanReach(parent) || (user.stat == SOFT_CRIT || UNCONSCIOUS))
```
Comment (ShizCalev) [inline code/datums/components/storage/concrete/wallet.dm:2]: if(user.incapacitated()) instead

also, update the parent proc so the check is in the same location

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/datums/components/storage/concrete/wallet.dm (hunk 2 above)
Comment (ivanmixo) [reply code/datums/components/storage/concrete/wallet.dm:2]: Good call, I didn't know that function existed.

File: code/datums/components/storage/storage.dm [hunk 3]
```diff
@@ -780,7 +780,7 @@
 	return hide_from(target)
 
 /datum/component/storage/proc/on_alt_click(datum/source, mob/user)
-	if(!isliving(user) || !user.CanReach(parent))
+	if(!isliving(user) || !user.CanReach(parent) || user.incapacitated())
 		return
 	if(locked)
 		to_chat(user, "<span class='warning'>[parent] seems to be locked!</span>")
```
Comment (ShizCalev) [inline code/datums/components/storage/storage.dm:786]: Remove the now redundant check at the bottom of the proc

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/datums/components/storage/storage.dm (hunk 3 above)
Comment (ivanmixo) [reply code/datums/components/storage/storage.dm:786]: Wait what do you want me to do? 

If I remove the user.incapacitated() from wallet.dm, but leave the user.incapacitated() in the parent proc in storage.dm the bug still persists. If I remove the user.incapacitated() from storage.dm the bug is still fixed, but it doesn't hurt to leave it in storage.dm, does it?

**Pull request merged**